.github/workflows/test-mac.yml

```
@@ -0,0 +1,88 @@
+---
+name: Test MacOS
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: >
+          Python version to currently test
+        type: string
+        required: true
+
+      install-optionals:
+        description: >
+          Decides whether we install optyional dependencies
+        type: boolean
+        default: false
+
+      runner:
+        description: >
+          Describes the system this workflow should run on.
+        type: string
+        required: true
+jobs:
+  tests-mac:
+    if: github.repository_owner == 'Qiskit'
+    name: ${{ inputs.runner }}-tests-Python-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 60
+    env:
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ inputs.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+          architecture: arm64
```

Is this correct? Shouldn't we only use the arm64 version of python on the macOS-14 runner?

---

You are correct! Do I make it do so conditionally? Like e.g.
```suggestion
          # This may not wokr
          architecture: ${{ inputs.runner == "macos-14" && "arm64" ||  "x64" }}
```

---

Does it work out of the box if you just remove this flag? Like I feel like the action should just use the native architecture if you don't specify anything.

---

I don't see any reason why it shouldn't work. It defaults to the architecture of the runner as per [documentation](https://github.com/actions/setup-python/tree/v5/?tab=readme-ov-file#supported-architectures).

---

Fixed in 4dc9eae